Source/Fuse.Controls.Primitives/TextControls/MobileTextEdit.uno

```diff
@@ -317,51 +327,43 @@ namespace Fuse.Controls
 				source.getPaddingBottom());
 			target.setTextScaleX(source.getTextScaleX());
 
-			/*
-				Nasty workaround to avoid Android rendering bug when textalignment is set to center,
-				doing it the normal way makes all the characters render on top of eachother...
-			*/
-			android.widget.GridLayout.LayoutParams lp = new android.widget.GridLayout.LayoutParams();
-			lp.rowSpec = android.widget.GridLayout.spec(0, android.widget.GridLayout.FILL);
-			int gravity = source.getGravity();
-			if ((gravity & android.view.Gravity.LEFT) == android.view.Gravity.LEFT)
-			{
-				lp.setGravity(android.view.Gravity.LEFT);
-				lp.columnSpec = android.widget.GridLayout.spec(0, android.widget.GridLayout.LEFT);
-			}
-			else if ((gravity & android.view.Gravity.RIGHT) == android.view.Gravity.RIGHT)
+			target.setLayoutParams(new android.widget.FrameLayout.LayoutParams(width, height));
+
+			if (android.os.Build.VERSION.SDK_INT >= 17)
+				target.setTextAlignment(android.view.View.TEXT_ALIGNMENT_GRAVITY);
+
+			target.setGravity(source.getGravity());
+
+			target.setHorizontallyScrolling(!isMultiline);
+
+			if (firstFrame)
```

Comment:
Based on the comment wouldn't it mean you need to set `firstFrame = true` whenever one of those properites changes as well?